pandas/core/nanops.py

```diff
@@ -786,7 +790,9 @@ def get_median(x, _mask=None):
         except ValueError as err:
             # e.g. "could not convert string to float: 'a'"
             raise TypeError(str(err)) from err
-    if mask is not None:
+    if not using_nan_sentinel and mask is not None:
+        if not values.flags.writeable:
+            values = values.copy()
```

Comment:
I think that in theory, this copy might not be needed, and we _could_ also ignore the read-only flag instead. The case that could still end up here is the masked Float64Dtype, where you do pass float64 data together with a mask. Although in that case, setting the values under the mask to NaN should I think always be "safe" at the moment (I don't think we already allow sharing data but not mask between FloatingArrays). But that's maybe something we can optimize later for Float64Dtype when we have a better idea of its memory model (and if sharing of data is definitely never possible)

Reply:
Now, looking a bit more, in practice we don't actually get here for FloatingArray at the moment, for two reasons:

1) we don't yet mark the numpy arrays as read-only when accessing those (FloatingArray `_data`), so it's only for the numpy float64 dtype that we would get read-only data at the moment

2) _if_ there are missing values in the FloatArray, we actually already copy the values inside `_get_values`: https://github.com/pandas-dev/pandas/blob/984d75543fce5c68f0dcc4f0b3500256e4a9c0c9/pandas/core/nanops.py#L312-L316

  Now, that seems double effort, and something we should fix, but that's for another issue (we are also setting with a fill_value of 0 here, which doesn't seem to make sense in the case of `median`, and then in nanmedian itself we either filter with the mask or set NaNs into it ..)


So I can also leave out this `if not values.flags.writeable:` check, because right now it will not be covered by any test / case in pandas, I think (unless you would use nanmedian directly)